routes/user.js

```diff
@@ -9,48 +10,80 @@ var router = express.Router();
-            req.body.passwordHash = authHelper.getPasswordHashSync(req.body.password, req.body.passwordSalt); // Get password hash
-            delete req.body.password; // Delete password permanently
-            log.info({req: req.body}, "EVENTUAL REQUEST");
-            return User.create(req.body).then(function (result) {
-                res.status(201).send(result);
-            }).catch(function (err) {
-                console.error(err);
+        req.body.approvingHash = makeLink(24);
+        req.body.passwordSalt = authHelper.generateSalt(16); // Create salt of 16 characters
+        req.body.passwordHash = authHelper.getPasswordHashSync(req.body.password, req.body.passwordSalt); // Get password hash
+        delete req.body.password; // Delete password permanently
+        req.body.isAdmin = false;
+        log.info({req: req.body}, "EVENTUAL REQUEST");
+        return User.create(req.body).then(function (result) {
+            nodemailer.createTestAccount().then(function () {
+                let transporter = nodemailer.createTransport({
+                    service: 'gmail',
+                    type: "SMTP",
+                    host: "smtp.gmail.com",
+                    secure: true,
+                    // Never fill this data in and add it to git! Only filled in locally or on the server!
+                    auth: {
+                        user: 'web@hsaconfluente.nl',
+                        pass: ''
+                    }
+                });
+                const link = "www.hsaconfluente.nl/api/user/approve/" + req.body.approvingHash;
+                transporter.sendMail({
+                    from: '"website" <web@hsaconfluente.nl>',
+                    to: req.body.email,
+                    subject: "Registration H.S.A. Confluente",
+                    text: "Thank you for making an account on our website hsaconfluente.nl! \n To fully activate your account, please visit this link: www.hsaconfluente.nl/api/user/approve/" + req.body.approvingHash,
+                    html: "Thank you for making an account on our website hsaconfluente.nl! To fully activate your account, please click <a href='" + link + "'>here!</a>"
+                }).then(function (info) {
+                    console.log("message sent: %s", info.messageId);
+                    console.log("Preview URL: %s", nodemailer.getTestMessageUrl(info));
+                })
             });
+            res.status(201).send(result);
+        }).catch(function (err) {
+            console.error(err);
         }).done();
     });
 
 router.route("/:id")
     .all(function (req, res, next) {
         var id = req.params.id;
         User.findByPk(req.params.id, {
-            attributes: ["id", "displayName", "email", "isAdmin"],
+            attributes: ["id", "firstName", "lastName", "displayName", "major", "track", "honorsGeneration", "campusCardNumber", "mobilePhoneNumber", "email", "isAdmin", "consentWithPortraitRight"],
```

Comment:
Should we not check more permissions before sending all this data to the frontend?

Reply:
Most checks of this are done in the functions after the .all function. However you might just want to check if user is logged in before requesting it from the db